reason-parser/src/reason_pprint_ast.ml

```diff
@@ -3375,8 +3399,8 @@ class printer  ()= object(self:'self)
     | PotentiallyLowPrecedence itm -> formatPrecedence ~loc:reducesAfterRight.pexp_loc itm
     | Simple itm -> itm
 
-  method ensureExpression expr ~reducesOnToken =
-    match self#unparseExprRecurse expr with
+  method ensureExpression ?(infixChain=false) ~reducesOnToken expr =
+    match self#unparseExprRecurse ~infixChain expr with
     | SpecificInfixPrecedence ({reducePrecedence; shiftPrecedence}, leftRecurse) ->
```

Comment:
Great! I think we could/should start extending that pattern through the rest of the printer. Imagine having a completely separate stage to manage all the stylistic/opinionated sections - separate from a stage that determines *correct* grouping according to precedence. It will be much easier for people to contribute fixes to stylistic formatting.